src/lib/formatter/RuleResultRecombinator.ts

```diff
@@ -344,10 +344,22 @@ URL: ${url}`;
 		return {columns, rows};
 	}
 
-	private static constructJson(results: RuleResult[]): string {
+	private static constructJson(results: RuleResult[], verboseViolations = false): string {
 		if (results.length === 0) {
 			return '';
 		}
+
+		if (verboseViolations) {
+			for (const result of results) {
+				for (const violation of result.violations) {
+					// in the json format we need to replace new lines in the message
+					// for the first line (ending with a colon) we will replace it with a space
+					// for following lines, we will replace it with a semicolon and a space
+					violation.message = violation.message.replace(/:\n/g, ': ').replace(/\n/g, '; ');
```

Comment:
You may wish to check what that does to non-RetireJS violations. I.e., scan something with both retirejs and eslint violations, and use verbose violations to see what the output looks like.
It's unlikely to cause any problems, but it's still probably worth taking a quick check.

Reply:
Good point, I am going to add a check so this only happens when the engine is retire-js, just to be safe.